### PR TITLE
reduce the amount of spelling issues / WORDLIST

### PR DESCRIPTION
**Pull request by m7pr**
The follow-up after https://github.com/insightsengineering/coredev-tasks/issues/235

Final `spelling::spell_check_package()`

We can not get rid of words that are in DESCRIPTION file, other than that mainly biological terms, or surnames or GitHub nicknames

```{R}
  WORD         FOUND IN
ADaM         description:3
ADLB         description:5
ADSL         description:4
balazs       g_correlationplot.Rd:302
             g_density_distribution_plot.Rd:154
             g_lineplot.Rd:300
             g_scatterplot.Rd:174
             t_summarytable.Rd:104
Balazs       g_boxplot.Rd:181
             g_correlationplot.Rd:302
             g_density_distribution_plot.Rd:154
             g_lineplot.Rd:300
             g_scatterplot.Rd:174
             t_summarytable.Rd:104
biomarker    g_boxplot.Rd:38,40,42,53
             g_correlationplot.Rd:57,59,61,65,67,81
             g_density_distribution_plot.Rd:36,38,40,44
             g_lineplot.Rd:44,46,48,50,52
             g_scatterplot.Rd:39,41,43,45,53
             g_spaghettiplot.Rd:43,45,47,49,51
             goshawk.Rd:6,8
             t_summarytable.Rd:23,25,27,37
             README.md:22
Biomarker    h_identify_loq_values.Rd:20
biomarkers   description:1
facetted     g_correlationplot.Rd:143
             g_density_distribution_plot.Rd:78
             g_scatterplot.Rd:90
             t_summarytable.Rd:29
facetting    g_correlationplot.Rd:101,103
             g_density_distribution_plot.Rd:58
             g_scatterplot.Rd:67,69
             NEWS.md:49,79,130
Forkers      README.md:50,60,62
jeffrey      g_boxplot.Rd:183
liu          g_lineplot.Rd:302
             g_spaghettiplot.Rd:230
LOQFL        description:6
nicholas     g_correlationplot.Rd:300
             g_density_distribution_plot.Rd:152
             g_scatterplot.Rd:172
             t_summarytable.Rd:102
npaszty      g_correlationplot.Rd:300
             g_density_distribution_plot.Rd:152
             g_scatterplot.Rd:172
             t_summarytable.Rd:102
paszty       g_correlationplot.Rd:300
             g_density_distribution_plot.Rd:152
             g_scatterplot.Rd:172
             t_summarytable.Rd:102
pre          goshawk.Rmd:54
quartile     g_lineplot.Rd:114
quartiles    g_boxplot.Rd:95,97
repo         README.md:9,15,58,62
roche        g_boxplot.Rd:183
             g_lineplot.Rd:302
             g_spaghettiplot.Rd:230
spacings     g_boxplot.Rd:100
tomlinsj     g_boxplot.Rd:183
tomlinson    g_boxplot.Rd:183
toth         g_correlationplot.Rd:302
             g_density_distribution_plot.Rd:154
             g_lineplot.Rd:300
             g_scatterplot.Rd:174
             t_summarytable.Rd:104
tothb        g_correlationplot.Rd:302
             g_density_distribution_plot.Rd:154
             g_scatterplot.Rd:174
             t_summarytable.Rd:104
wenyi        g_lineplot.Rd:302
             g_spaghettiplot.Rd:230
```

### CODE REVIEW

File: R/g_boxplot.R [hunk 1]
```diff
@@ -10,35 +10,35 @@
...
 #'   if the default limits are not suitable.
-#' @param facet_var variable to facet the plot by, or "None" if no faceting
+#' @param facet_var variable to facet the plot by, or `"None"` if no faceting
 #'   required.
 #' @param xaxis_var variable used to group the data on the x-axis.
-#' @param facet_ncol number of facets per row.  NULL = Use the default for ggplot2::facet_wrap
+#' @param facet_ncol number of facets per row.  NULL = Use the default for `ggplot2::facet_wrap`
 #' @param rotate_xlab 45 degree rotation of x-axis label values.
-#' @param font_size point size of tex to use.  NULL is use default size
+#' @param font_size point size of `tex` to use.  NULL is use default size
```
Comment (chlebowa) [inline R/g_boxplot.R:34]: this should be "text", I suppose

File: R/g_boxplot.R [hunk 2]
```diff
@@ -10,35 +10,35 @@
...
 #' @param unit biomarker unit label e.g. (U/L)
-#' @param color_manual vector of color for trt_group
-#' @param shape_manual vector of shapes (used with loq_flag_var)
+#' @param color_manual vector of color for `trt_group`
+#' @param shape_manual vector of shapes (used with `loq_flag_var`)
 #' @param box add boxes to the plot (boolean)
 #' @param ylim ('numeric vector') optional, a vector of length 2 to specify the minimum and maximum of the y-axis
 #'   if the default limits are not suitable.
-#' @param facet_var variable to facet the plot by, or "None" if no faceting
+#' @param facet_var variable to facet the plot by, or `"None"` if no faceting
```
Comment (chlebowa) [inline R/g_boxplot.R:29]: why would "None" be printed as code?

File: R/g_boxplot.R [hunk 3]
```diff
@@ -10,35 +10,35 @@
 #' of the box indicate the degree of dispersion (spread) and skewness in the
 #' data, and show outliers. In addition to the points themselves, they allow one
 #' to visually estimate various L-estimators, notably the interquartile range,
-#' midhinge, range, mid-range, and trimean.
+#' `midhinge`, range, mid-range, and `trimean`.
```
Comment (chlebowa) [inline R/g_boxplot.R:13]: Either backtick all estimators or none.
Would these actually not belong in the WORDLIST?

File: R/g_boxplot.R [hunk 4]
```diff
@@ -263,7 +263,7 @@ g_boxplot <- function(data,
 
   # Add facetting.
   if (!is.null(facet_var)) {
-    if (facet_var != "None" & facet_var %in% names(data)) {
+    if (facet_var != "None" && facet_var %in% names(data)) {
```
Comment (chlebowa) [inline R/g_boxplot.R:266]: I think the original `&` was appropriate.

File: R/geom_axes_line.R [hunk 5]
```diff
@@ -1,6 +1,6 @@
 #' Add horizontal and/or vertical lines and their legend labels to a plot
 #'
-#' This function is currently designed to be used with ('g_boxplot'), ('g_correlationplot'), ('g_spaghettiplot'),
+#' This function is currently designed to be used with (`'g_boxplot'`), (`'g_correlationplot'`), (`'g_spaghettiplot'`),
```
Comment (chlebowa) [inline R/geom_axes_line.R:3]: remove `'` signs

File: R/t_summarytable.R [hunk 6]
```diff
@@ -2,16 +2,16 @@
 #'
 #' Output descriptive summary statistics table as a data frame. Includes biomarker, treatment,
 #' visit,
-#' n, mean, median, sd, min, max, %missing values, % LOQ values.
+#' n, mean, median, SD, min, max, %missing values, % `LOQ` values.
```
Comment (chlebowa) [inline R/t_summarytable.R:5]: Is capitalization of "sd" done solely because the linter accepts "SD" and not "sd"?

**Review by chlebowa — COMMENTED** [review]
I am not convinced representing any non-standard word as code it the way to go. It is fine for function or parameter names but I believe statistical term and acronyms should be printed as normal text.

File: R/g_boxplot.R (hunk 2 above)
Comment (pawelru) [reply R/g_boxplot.R:29]: IMHO the quotation indicates character type of object whereas no-quotation indicates unknown object (incl. function), e.g. let's have the following `foo_factory <- function(foo_fun = foo, name = "test") {...}` and when documenting it I can state that `foo_fun` takes `foo` as a default (i.e. no quote) and the default for `name` argument is `"test"` (quoted). So I do agree with this change, but... (see below)

Yet another thing is whether this is correct. Something close to `None` looked very pythonish and I would double-check if this should have been state as being empty.

File: R/g_boxplot.R (hunk 4 above)
Comment (pawelru) [reply R/g_boxplot.R:266]: good catch
this indeed change the logic which is out of the scope of this PR
In general I have nothing against fixing bugs as we spot them but this should be noted in PR and re-tested during PR so reviewer needs to be aware of this.

File: R/geom_axes_line.R (hunk 5 above)
Comment (pawelru) [reply R/geom_axes_line.R:3]: or even better: convert to markdown references to other documented objects

File: R/g_boxplot.R (hunk 2 above)
Comment (chlebowa) [reply R/g_boxplot.R:29]: Fair enough, code it is.

`if (facet_var != "None" & facet_var %in% names(data)) {` suggests that the "None" literal is just a way to specify one doesn't want faceting. Although I I'm not entirely sure why it is only invoked `if (!is.null(facet_var)) {`.

File: R/g_boxplot.R (hunk 1 above)
Comment (m7pr) [reply R/g_boxplot.R]: 9cad69e71a18297e4df9b431b5675508e5f636bc

File: R/geom_axes_line.R (hunk 5 above)
Comment (m7pr) [reply R/geom_axes_line.R]: a7530871a6aba2381de651ce1f5287a4196f3609

File: R/t_summarytable.R (hunk 6 above)
Comment (m7pr) [reply R/t_summarytable.R:5]: @chlebowa this is a statistical term and it has official capitalized short-form. You can see a lot of statistical terms being written in this manner, like AIC, PCA, PCR, MLE. And yes, those are omitted by `spelling::spell_check_package`
```{R}
hunspell::en_stats
[1] "Accessor"           "accessor"           "accessors"          "ACF"                "Affymetrix"        
  [6] "agglomerative"      "AIC"                "Akaike"             "Akaike's"           "alikes"            
 [11] "ANOVA"              "API"                "approximative"      "ARIMA"              "ARMA"              
 [16] "ascii"              "AUC"                "autocorrelation"    "autocorrelations"   "autocovariance"    
 [21] "autocovariances"    "autoregression"     "autoregressive"     "autosomal"          "backend"           
 [26] "backfitting"        "backquote"          "backquoted"         "backreference"      "backticks"         
 [31] "barplot"            "barplots"           "basename"           "BFGS"               "BibTeX"            
 [36] "bilinear"           "binomially"         "Bioconductor"       "biplot"             "biplots"           
 [41] "bitwise"            "Bivariate"          "bivariate"          "biweight"           "boolean"           
 [46] "boxplot"            "boxplots"           "breakpoint"         "capita"             "cartesian"         
 [51] "CCA"                "CDF"                "CDFs"               "checkbox"           "Cholesky"          
 [56] "CIs"                "CJK"                "clusterings"        "CMYK"               "codepage"          
 [61] "codings"            "collinearity"       "colnames"           "columnwise"         "componentwise"     
 [66] "cophenetic"         "correlation"        "covariables"        "covariance"         "covariances"       
 [71] "Covariate"          "covariate"          "Covariates"         "covariates"         "Cp"                
 [76] "CPUs"               "CRAN"               "criterium"          "CRLF"               "CSS"               
 [81] "CSV"                "cumulated"          "cutpoint"           "cutpoints"          "datafile"          
 [86] "Dataframe"          "dataframe"          "dataframes"         "datapoints"         "Dataset"           
 [91] "dataset"            "Datasets"           "datasets"           "DBI"                "DCF"               
 [96] "dendrogram"         "dendrograms"        "deparse"            "deparsed"           "deparses"          
[101] "deparsing"          "detrended"          "devel"              "deviances"          "DIC"               
[106] "dichotomized"       "differenced"        "differentiable"     "differentially"     "dimensionality"    
[111] "dimnames"           "discretization"     "discretize"         "discretized"        "DLL"               
[116] "DLLs"               "EDF"                "eigenvector"        "eigenvectors"       "EISPACK"           
[121] "elementwise"        "encodings"          "endian"             "EPS"                "equispaced"        
[126] "ESRI"               "Euclidian"          "exceedance"         "executables"        "expectation"       
[131] "exponentiated"      "FFT"                "Filename"           "filename"           "filenames"         
[136] "filepaths"          "FIPS"               "forecasted"         "Fortran"            "FreeBSD"           
[141] "frequentist"        "functionalities"    "functionals"        "GAM"                "GARCH"             
[146] "gaussian"           "Gaussians"          "GCV"                "genomic"            "genotype"          
[151] "genotyped"          "genotypes"          "genotypic"          "genotyping"         "Ghostscript"       
[156] "gif"                "GIS"                "GLM"                "GLMs"               "Google"            
[161] "GPD"                "GPL"                "grob"               "grobs"              "GUIs"              
[166] "haplotype"          "haplotypes"         "heatmap"            "Helvetica"          "heteroscedastic"   
[171] "heteroscedasticity" "hoc"                "homogenous"         "homoscedastic"      "HPD"               
[176] "HSV"                "html"               "HWE"                "hypergeometric"     "hyperparameter"    
[181] "hyperparameters"    "identifiability"    "IEEE"               "iff"                "iid"               
[186] "individual's"       "informations"       "inline"             "internet"           "interquartile"     
[191] "invariance"         "invertibility"      "IQR"                "IRT"                "IWLS"              
[196] "Jacobian"           "JASA"               "jittered"           "jittering"          "JPEG"              
[201] "Kolmogorov"         "kriging"            "Kruskal"            "Kurtosis"           "kurtosis"          
[206] "LAPACK"             "LaTeX"              "latin"              "LDA"                "LGPL"              
[211] "LHS"                "lhs"                "linearization"      "LINPACK"            "localhost"         
[216] "locator"            "logicals"           "logit"              "logits"             "loglikelihood"     
[221] "loglinear"          "lognormal"          "lookup"             "lowess"             "LRT"               
[226] "Mahalanobis"        "manhattan"          "MANOVA"             "MATLAB"             "Matlab"            
[231] "matrixes"           "maxima"             "MCD"                "MCMC"               "MDI"               
[236] "medoids"            "metacharacters"     "metadata"           "microarray"         "microarrays"       
[241] "microsatellite"     "minima"             "minimax"            "misclassification"  "missingness"       
[246] "missings"           "MLE"                "MLEs"               "monospaced"         "MPI"               
[251] "MSE"                "multi"              "multibyte"          "multicore"          "multiline"         
[256] "multilocus"         "multimodal"         "Multinomial"        "multinomial"        "multinomially"     
[261] "namespace"          "namespaces"         "NAs"                "NCBI"               "nonlinearity"      
[266] "nonnegative"        "Nonparametric"      "nonparametric"      "numerics"           "OECD"              
[271] "ok"                 "OLS"                "Omegahat"           "one's"              "online"            
[276] "orthogonalized"     "outlier"            "outliers"           "overdispersed"      "overdispersion"    
[281] "overfitting"        "overlayed"          "overplotting"       "pairlist"           "pairlists"         
[286] "parallelized"       "parameterisation"   "parameterization"   "parameterizations"  "parameterize"      
[291] "parameterized"      "parametrization"    "parametrizations"   "pathname"           "patient's"         
[296] "PCA"                "PCR"                "PCRE"               "PDF"                "pdf"               
[301] "periodogram"        "phenotypic"         "Phylogenetic"       "phylogenetic"       "piecewise"         
[306] "PLS"                "plugin"             "PNG"                "pointsize"          "pointwise"         
[311] "polytomous"         "POSIX"              "PostScript"         "precalculated"      "prepend"           
[316] "prepended"          "preprocessing"      "prespecified"       "probit"             "pseudomedian"      
[321] "QQ"                 "QTL"                "Quantile"           "quantile"           "Quantiles"         
[326] "quantiles"          "randomizations"     "rbind"              "reals"              "regressor"         
[331] "regressors"         "REML"               "reparameterized"    "resample"           "resampled"         
[336] "resamples"          "Resampling"         "resampling"         "resize"             "resized"           
[341] "reweighted"         "RGB"                "rhs"                "RMSE"               "RNG"               
[346] "ROC"                "rownames"           "RSS"                "runtime"            "SANN"              
[351] "SAS"                "scalings"           "Scatterplot"        "scatterplot"        "scatterplots"      
[356] "screeplot"          "semidefinite"       "semiparametric"     "shapefile"          "sib"               
[361] "significand"        "skewness"           "Smirnov"            "smoothers"          "SNP"               
[366] "SNPs"               "Solaris"            "Spearman"           "Spearman's"         "spinogram"         
[371] "spinograms"         "Splus"              "SPSS"               "SQL"                "sqrt"              
[376] "startup"            "stationarity"       "stepsize"           "stepwise"           "stochastically"    
[381] "studentized"        "subdirectories"     "subdirectory"       "subexpression"      "subexpressions"    
[386] "subgraph"           "subintervals"       "sublist"            "sublists"           "submatrices"       
[391] "submatrix"          "submodel"           "submodels"          "subpopulations"     "subsample"         
[396] "subsamples"         "subsampling"        "subsequence"        "subseries"          "subsetted"         
[401] "Subsetting"         "subsetting"         "substring"          "substrings"         "subtree"           
[406] "subtrees"           "superclass"         "superclasses"       "superset"           "supremum"          
[411] "SVD"                "SVG"                "SVM"                "Sweave"             "taxa"              
[416] "taxon"              "Tcl"                "TCP"                "TeX's"              "thresholded"       
[421] "thresholding"       "tickmarks"          "timepoints"         "timestamp"          "timestep"          
[426] "Tk"                 "TODO"               "tradeoff"           "Tukey"              "Tukey's"           
[431] "tuple"              "txt"                "unary"              "unexported"         "unimodal"          
[436] "uniquenesses"       "univariate"         "unix"               "unlisting"          "unnormalized"      
[441] "unpenalized"        "unstandardized"     "url"                "URW"                "USD"               
[446] "username"           "UTC"                "UTF"                "utils"              "UTM"               
[451] "variable's"         "variance"           "variogram"          "vertices"           "viewport"          
[456] "viewport's"         "viewports"          "voxel"              "webpage"            "Weibull"           
[461] "whitespace"         "Wilcoxon"           "wildcard"           "wildcards"          "wireframe"         
[466] "Wishart"            "wrt"                "x's"                "xml"
```

File: R/g_boxplot.R (hunk 4 above)
Comment (m7pr) [reply R/g_boxplot.R:266]: I think I remember this as one of the requirements of the current R CMD CHECK (to use `&&` and `||`) in R 4.3.0 but can not reproduce it on the `main` branch when triggering tests


`&&` seems superior to `&` and does less operations in `LHS && RHS` if `LHS` is FALSE. 
if `LHS` is TRUE it does the same as `&` , but when `LHS` is FALSE it stops and does not check any further 

let me revert this change for now

**Comment by m7pr** [comment]
Thanks @chlebowa for comments and the review! Appreciate as always! Did a couple of updates to resolve your comments and suggestions.

About the below

> I am not convinced representing any non-standard word as code it the way to go. It is fine for function or parameter names but I believe statistical term and acronyms should be printed as normal text.

This is a challenging process to unify multiple packages written by various developers across a couple of past years. I needed to create one form of quoting for more than one repo. I decided that we should quote packages, functions, parameter names, objects and data structure names. I reckon PARAMCD or ADaM/ARM/ADSL are column names or dataset types specific to our are, that's why I decided to quote those, especially because in some other packages it was already quoted. 

In the manual page this quoting change those terms to a text written as CODE which indicates this is an R-specific object or a function or a parameter, making you realize that under the acronim or short-code like ADaM there is an actual structure that is represented by R objects.

image

goshawk is the last one having this WORDLIST correction - but the same was applied (or rather unified) in other packages as well.

Here we do not have much statistical terms, but I was trying also to highlight them, especially if they are a-quoted-character strings like `"wald"` that states for the possible value of the parameter. Those gets quoted as code as they represent a character string.  

I hope that makes it more understandable and bring more motivation to the changes I proposed.

File: R/g_boxplot.R (hunk 4 above)
Comment (m7pr) [reply R/g_boxplot.R:266]: yes, but I checked `facet_var` was length one in this case so it's the same for && and & for objects of length 1. anyway I reverted this as I don't see that in R CMD CHECK notes anymore

Comment (chlebowa) [reply R/g_boxplot.R:266]: Upon second thought I think my initial comment was wrong. Sorry for creating confusion.

On a slightly unrelated note, I am not sure why `facet_var` has to be checked for NULL and then for "None". If the argument defaults to NULL and then no faceting is applied, why add special handling for "None"?

Comment (m7pr) [reply R/g_boxplot.R:266]: This actually brings this error on LINTR checks - https://github.com/insightsengineering/goshawk/blob/222_decouple_scda%40main/R/g_boxplot.R#L264
I have added #nolint in other PR 34194d2d9ec564fb51c20470df7da709195fe34c to omit that

**Review by chlebowa — APPROVED** [review]

**Pull request merged**